File: app/src/main/java/com/bignerdranch/stockwatcher/util/ProgressDialogFragment.java
package com.bignerdranch.stockwatcher.util;

import android.app.Dialog;
import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

public class ProgressDialogFragment extends DialogFragment {

    private static final String ARG_MESSAGE = "message";

    public static ProgressDialogFragment newInstance(String message) {
        Bundle args = new Bundle();
        args.putString(ARG_MESSAGE, message);
        ProgressDialogFragment fragment = new ProgressDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        ProgressDialog dialog = new ProgressDialog(getActivity());
        String message = getArguments().getString(ARG_MESSAGE);
        dialog.setMessage(message);
        dialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
        return dialog;
    }
}

File: app/src/main/java/com/bignerdranch/stockwatcher/util/ProgressDialogFragment.kt
package com.bignerdranch.stockwatcher.util

import android.app.Dialog
import android.app.ProgressDialog
import android.os.Bundle
import android.support.v4.app.DialogFragment

class ProgressDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = ProgressDialog(context)
        val message = arguments.getString(ARG_MESSAGE)
        dialog.setMessage(message)
        dialog.setProgressStyle(ProgressDialog.STYLE_SPINNER)
        return dialog
    }

    companion object {

        private val ARG_MESSAGE = "message"

        fun newInstance(message: String): ProgressDialogFragment {
            val args = Bundle()
            args.putString(ARG_MESSAGE, message)
            val fragment = ProgressDialogFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
